e2e/_suites/fleet/services.go

```diff
@@ -270,32 +385,196 @@ func newDebianInstaller() (ElasticAgentInstaller, error) {
 
 	return ElasticAgentInstaller{
 		artifactArch:      arch,
 		artifactExtension: extension,
 		artifactName:      artifact,
 		artifactOS:        os,
 		artifactVersion:   version,
+		binDir:            binDir,
 		commitFile:        ".elastic-agent.active.commit",
+		EnrollFn:          enrollFn,
 		homeDir:           "/etc/elastic-agent/",
 		image:             image,
-		InstallCmds:       []string{"apt", "install", "/" + binaryName, "-y"},
-		logDir:            "/var/lib/elastic-agent/data/elastic-agent-%s/logs/",
+		InstallFn:         installFn,
+		installerType:     "deb",
 		logFile:           "elastic-agent-json.log",
 		name:              binaryName,
 		path:              binaryPath,
-		PostInstallFn:     fn,
+		PostInstallFn:     postInstallFn,
+		PreInstallFn:      preInstallFn,
 		processName:       ElasticAgentProcessName,
 		profile:           profile,
 		service:           service,
 		tag:               tag,
+		UninstallFn:       unInstallFn,
+		workingDir:        binDir + "logs/",
 	}, nil
 }
 
+// newTarInstaller returns an instance of the Debian installer
+func newTarInstaller(image string, tag string) (ElasticAgentInstaller, error) {
+	image = image + "-systemd" // we want to consume systemd boxes
+	service := image
+	profile := FleetProfileName
+
+	// extract the agent in the box, as it's mounted as a volume
+	artifact := "elastic-agent"
+	version := agentVersion
+	os := "linux"
+	arch := "x86_64"
+	extension := "tar.gz"
+
+	tarFile, binaryPath, err := downloadAgentBinary(artifact, version, os, arch, extension)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"artifact":  artifact,
+			"version":   version,
+			"os":        os,
+			"arch":      arch,
+			"extension": extension,
+			"error":     err,
+		}).Error("Could not download the binary for the agent")
+		return ElasticAgentInstaller{}, err
+	}
+
+	commitFile := ".elastic-agent.active.commit"
+	homeDir := "/elastic-agent/"
+	binDir := "/usr/bin/"
+
+	preInstallFn := func() error {
+		commitFile := homeDir + commitFile
+		return installFromTar(profile, image, service, tarFile, commitFile, artifact, version, os, arch)
+	}
+	installFn := func(containerName string, token string) error {
+		hash, err := getElasticAgentHash(containerName, homeDir+commitFile)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"containerName": containerName,
+				"error":         err,
+			}).Error("Could not get agent hash in the container")
+
+			return err
+		}
+
+		// install the elastic-agent to /usr/bin/elastic-agent using command
+		binary := fmt.Sprintf("/elastic-agent/data/elastic-agent-%s/", hash) + artifact
+		args := []string{"--force", "--insecure", "--enrollment-token", token, "--kibana-url", "http://kibana:5601"}
+
+		err = runElasticAgentCommand(profile, image, service, binary, "install", args)
+		if err != nil {
+			return fmt.Errorf("Failed to install the agent with subcommand: %v", err)
+		}
+		return nil
+	}
+	enrollFn := func(token string) error {
```

Comment:
enroll is performed out of the box using install so this should not be necessary

Reply:
If you take a look, we are providing the enroll command for further usage in the re-enroll scenario. As defined in the life cycle (see PR description), we use install command to initial enroll, but enroll for following ones